library/src/main/java/com/squareup/timessquare/CalendarPickerView.java

```diff
@@ -629,6 +629,58 @@ public boolean selectDate(Date date, boolean smoothScroll) {
     return wasSelected;
   }
 
+  /**
+   * Select a new range of dates. If you are in {@link SelectionMode#MULTIPLE}, the new dates will
+   * be added to the list of selected dates.
+   * <p>
+   * If the selection was made (selectable date, in range), the view will scroll to first date.
+   *
+   * @return - whether we were able to set the dates
+   */
+    public boolean selectDates(Collection<Date> selectedDates) {
+      return selectDates(selectedDates, false);
+  }
+
+  /**
+   * Select a new range of dates. If you are in {@link SelectionMode#MULTIPLE}, the new dates will
+   * be added to the list of selected dates.
+   * <p>
+   * If the selection was made (selectable date, in range), the view will scroll to first date.
+   *
+   * @return - whether we were able to set the dates
+   */
+  public boolean selectDates(Collection<Date> selectedDates, boolean smoothScroll) {
+    boolean firstDateThatWasSelected = false;
+    int firstMonthIndex = 1;
+    int monthIndex = 1;
+
+    if (selectedDates != null) {
+      for (Date date : selectedDates) {
+        validateDate(date);
+
+        MonthCellWithMonthIndex monthCellWithMonthIndex = getMonthCellWithIndexByDate(date);
+
+        if (monthCellWithMonthIndex == null || !isDateSelectable(date)) {
+          return false;
+        }
+
+        monthIndex = monthCellWithMonthIndex.monthIndex;
+
+        if (!firstDateThatWasSelected) {
```

Comment:
This doesn't look right.  You're only going to call `doSelectDate` on the first one that gets passed in.

Please add a test to `CalendarPickerViewTest` to make sure this all works as expected.